File: scripts/generate-icons.js
#!/usr/bin/env node

/**
 * Generate properly formatted Android adaptive icons with correct safe zones.
 * Requires: npm install sharp
 */

const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const ICON_SIZE = 1080;
const PADDING_RATIO = 0.25; // 25% padding on each side = 50% total
const SAFE_ZONE_SIZE = Math.floor(ICON_SIZE * (1 - 2 * PADDING_RATIO));
const OFFSET = Math.floor(ICON_SIZE * PADDING_RATIO);

async function createAdaptiveIcon() {
  try {
    const assetsDir = path.join(__dirname, '..', 'assets', 'images');
    const sourceIcon = path.join(assetsDir, 'icon.png');
    const foregroundPath = path.join(assetsDir, 'android-icon-foreground.png');

    if (!fs.existsSync(sourceIcon)) {
      console.error(`✗ Source icon not found: ${sourceIcon}`);
      process.exit(1);
    }

    console.log('Generating Android adaptive icons...');
    console.log('='.repeat(50));

    // Create transparent background
    const transparentBg = Buffer.alloc(ICON_SIZE * ICON_SIZE * 4);

    // Resize source icon to fit in safe zone
    const resizedIcon = await sharp(sourceIcon)
      .resize(SAFE_ZONE_SIZE, SAFE_ZONE_SIZE, {
        fit: 'contain',
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      })
      .png()
      .toBuffer();

    // Create the adaptive icon by compositing
    await sharp({
      create: {
        width: ICON_SIZE,
        height: ICON_SIZE,
        channels: 4,
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      }
    })
      .composite([
        {
          input: resizedIcon,
          top: OFFSET,
          left: OFFSET
        }
      ])
      .png()
      .toFile(foregroundPath);

    console.log(`✓ Created: ${foregroundPath}`);
    console.log(`  Size: ${ICON_SIZE}x${ICON_SIZE}`);
    console.log(`  Safe zone: ${SAFE_ZONE_SIZE}x${SAFE_ZONE_SIZE}`);
    console.log(`  Padding: ${OFFSET}px on each side`);
    console.log();

    // Create monochrome version
    const monochromeIcon = await sharp(sourceIcon)
      .resize(ICON_SIZE, ICON_SIZE, {
        fit: 'contain',
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      })
      .modulate({
        saturation: 0 // Convert to grayscale
      })
      .png()
      .toBuffer();

    const monochromePath = path.join(assetsDir, 'android-icon-monochrome.png');
    await sharp(monochromeIcon)
      .toFile(monochromePath);

    console.log(`✓ Created monochrome: ${monochromePath}`);
    console.log();
    console.log('='.repeat(50));
    console.log('✓ Icon generation complete!');
    console.log();
    console.log('Next steps:');
    console.log('1. Rebuild the app: eas build --platform android');
    console.log('2. Test on Android device');
    console.log('3. Check app drawer - icon should display correctly');

  } catch (error) {
    console.error('✗ Error generating icons:', error.message);
    process.exit(1);
  }
}

// Run if sharp is available
try {
  require.resolve('sharp');
  createAdaptiveIcon();
} catch (e) {
  console.error('✗ sharp is not installed');
  console.error('Install it with: npm install sharp');
  process.exit(1);
}

File: scripts/generate-icons.py
#!/usr/bin/env python3
"""
Generate properly formatted Android adaptive icons with correct safe zones.
This script takes the main icon and creates versions with proper padding.
"""

from PIL import Image
import os

def create_adaptive_icon(source_path, output_path, size=1080, padding_ratio=0.25):
    """
    Create an adaptive icon with proper safe zone padding.
    
    Args:
        source_path: Path to source icon image
        output_path: Path to save the adaptive icon
        size: Output size (default 1080x1080)
        padding_ratio: Padding ratio (0.25 = 25% padding on each side)
    """
    try:
        # Open the source image
        img = Image.open(source_path)
        
        # Convert to RGBA if needed
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        # Create a new image with transparent background
        adaptive_icon = Image.new('RGBA', (size, size), (0, 0, 0, 0))
        
        # Calculate the safe zone (inner area where content should be)
        safe_zone_size = int(size * (1 - 2 * padding_ratio))
        
        # Resize the source image to fit in the safe zone
        img_resized = img.resize((safe_zone_size, safe_zone_size), Image.Resampling.LANCZOS)
        
        # Calculate position to center the image
        offset = int(size * padding_ratio)
        
        # Paste the resized image onto the adaptive icon
        adaptive_icon.paste(img_resized, (offset, offset), img_resized)
        
        # Save the result
        adaptive_icon.save(output_path, 'PNG')
        print(f"✓ Created: {output_path}")
        print(f"  Size: {size}x{size}")
        print(f"  Safe zone: {safe_zone_size}x{safe_zone_size}")
        print(f"  Padding: {int(size * padding_ratio)}px on each side")
        
    except Exception as e:
        print(f"✗ Error creating adaptive icon: {e}")

def create_monochrome_icon(source_path, output_path, size=1080):
    """
    Create a monochrome version of the icon (white on transparent).
    """
    try:
        img = Image.open(source_path)
        
        # Convert to RGBA
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        # Create monochrome version (white)
        data = img.getdata()
        new_data = []
        
        for item in data:
            # If pixel has any opacity, make it white
            if item[3] > 0:  # If alpha > 0
                new_data.append((255, 255, 255, item[3]))
            else:
                new_data.append((0, 0, 0, 0))
        
        img.putdata(new_data)
        
        # Resize to target size
        img = img.resize((size, size), Image.Resampling.LANCZOS)
        
        img.save(output_path, 'PNG')
        print(f"✓ Created monochrome: {output_path}")
        
    except Exception as e:
        print(f"✗ Error creating monochrome icon: {e}")

def main():
    """Generate all icon variants."""
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    assets_dir = os.path.join(base_dir, 'assets', 'images')
    
    source_icon = os.path.join(assets_dir, 'icon.png')
    
    if not os.path.exists(source_icon):
        print(f"✗ Source icon not found: {source_icon}")
        return
    
    print("Generating Android adaptive icons...")
    print("=" * 50)
    
    # Create adaptive icon foreground with proper padding
    foreground_path = os.path.join(assets_dir, 'android-icon-foreground.png')
    create_adaptive_icon(source_icon, foreground_path, size=1080, padding_ratio=0.25)
    
    print()
    
    # Create monochrome version
    monochrome_path = os.path.join(assets_dir, 'android-icon-monochrome.png')
    create_monochrome_icon(source_icon, monochrome_path, size=1080)
    
    print()
    print("=" * 50)
    print("✓ Icon generation complete!")
    print()
    print("Next steps:")
    print("1. Rebuild the app: eas build --platform android")
    print("2. Test on Android device")
    print("3. Check app drawer - icon should display correctly")

if __name__ == '__main__':
    main()
